lib/collection/request-auth.js

```diff
@@ -180,7 +188,7 @@ _.extend(RequestAuth, /** @lends RequestAuth */ {
         }
 
         // Create intermediate function to instantiate a new handler with the definition in prototype chain
-        var AuthHandler = function AuthHandler () { },
```

Comment:
@kunagpal One drawback of replacing this particular function is that we're losing the function name. That is a problem because we will not be able to distinguish between this in heapdumps (heapdumps only show constructor).